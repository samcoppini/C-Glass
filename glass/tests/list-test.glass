{M
    [(printList)
        "List length: "oo.?
        (_len)l(getLength).?=
        (_len)*o(onl).?

        "List contents:"oo.?

        (_a)A!
        (_i)<0>=
        (_cmp)(_i)*(_len)*(_a)(lt).?=
        /(_cmp)
            (_i)*l(get).?
            " "oo.?o(on).?

            (_i)(_i)*<1>(_a)a.?=
            (_cmp)(_i)*(_len)*(_a)(lt).?=
        \

        "\n"oo.?
    ]

    [m
        t$
        o(Output)!
        l(List)!

        "Length of empty list: "oo.?
        l(getLength).?o(onl).?

        "Adding 42 to the list"o(ol).?
        <42>l(add).?
        t(printList).?

        "Adding 54 to the list"o(ol).?
        <54>l(add).?
        t(printList).?

        "Adding -7 to the list"o(ol).?
        <-7>l(add).?
        t(printList).?

        "Adding 99 to the list"o(ol).?
        <99>l(add).?
        t(printList).?

        "Adding 0 to the list"o(ol).?
        <0>l(add).?
        t(printList).?
    ]
}