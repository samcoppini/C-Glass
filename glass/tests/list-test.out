Length of empty list: 0
Adding 42 to the list
List length: 1
List contents: 42
Adding 54 to the list
List length: 2
List contents: 42 54
Adding -7 to the list
List length: 3
List contents: 42 54 -7
Adding 99 to the list
List length: 4
List contents: 42 54 -7 99
Adding 0 to the list
List length: 5
List contents: 42 54 -7 99 0
